### PR TITLE
ThemeManager - Adding deprecation warning for 'setTheme()'.

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
ThemeManager - Adding deprecation warning for 'setTheme()' to start the deprecation process before removing the method on the next major version.
Removing all usages of ThemeManager global constants - 'primaryColor', 'CTA', 'titleColor' etc.

## Changelog
ThemeManager - Starting deprecation process for 'setTheme()' method.

### CODE REVIEW

File: src/style/themeManager.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export class ThemeManager {
   }
 
   setTheme(overrides: Dictionary<string>) {
+    console.warn('ThemeManager.setTheme() is deprecated. Please remove usage before next uilib major version update.');
```
Comment (ethanshar) [inline src/style/themeManager.ts:28]: I would also add:
"Consider using ThemeManager.setComponentTheme instead"

**Review by ethanshar — CHANGES_REQUESTED** [review]

File: src/style/themeManager.ts (hunk 1 above)
Comment (Inbal-Tish) [reply src/style/themeManager.ts]: Done.

**Comment by Inbal-Tish** [comment]
@ethanshar I added to this PR the removal of all usages of the ThemeManager global constants.

File: src/style/themeManager.ts (hunk 1 above)
Comment (Inbal-Tish) [reply src/style/themeManager.ts]: Done.

File: src/components/touchableOpacity/index.tsx [hunk 2]
```diff
@@ -62,6 +62,11 @@ type Props = BaseComponentInjectedProps &
 class TouchableOpacity extends PureComponent<Props, {active: boolean}> {
   static displayName = 'TouchableOpacity';
 
+  static defaultProps = {
+    throttleTime: 0,
+    throttleOptions: {leading: true, trailing: false}
+  }
```
Comment (ethanshar) [inline src/components/touchableOpacity/index.tsx:68]: This will override our config in private uilib.. 
Why did you add it?

Comment (Inbal-Tish) [reply src/components/touchableOpacity/index.tsx:68]: This is to replace the configuration in ThemeManager (line #16).

Comment (ethanshar) [reply src/components/touchableOpacity/index.tsx:68]: Ohh I see now..
The thing is that setting it as `defaultProps` give it a different priority. 
Unfortunately, defaultProps are prioritized over theme props and override them. 
And the way it was defined before was considered as theme props..

Another option we can do is to use default Constants like we do sometimes when we extract the values 
WDYT?

Comment (Inbal-Tish) [reply src/components/touchableOpacity/index.tsx:68]: I bit messy but I guess that could be a solution. I'll push a fix

Comment (Inbal-Tish) [reply src/components/touchableOpacity/index.tsx]: Done

**Review by ethanshar — APPROVED** [review]
